diff --git a/dist/calendar/calendarDate.js b/dist/calendar/calendarDate.js
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/dist/calendar/calendarDate.js
+++ b/dist/calendar/calendarDate.js
@@ -178,7 +178,7 @@
  * encode(Calendar.make(2026, 4, 13)) // "2026-04-13"
  * ```
  */
-export const CalendarDateFromIsoString = S.String.pipe(S.decodeTo(CalendarDate, SchemaTransformation.transformOrFail({
+export const CalendarDateFromIsoString = S.String.pipe(S.decodeTo(CalendarDate, SchemaTransformation.transformEffect({
     decode: input => {
         const match = input.match(isoPattern);
         if (match === null) {
diff --git a/dist/url/index.js b/dist/url/index.js
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/dist/url/index.js
+++ b/dist/url/index.js
@@ -17,7 +17,7 @@
         port: S.String,
     }),
 });
-const LocationAndHrefFromString = S.String.pipe(S.decodeTo(LocationAndHref, SchemaTransformation.transformOrFail({
+const LocationAndHrefFromString = S.String.pipe(S.decodeTo(LocationAndHref, SchemaTransformation.transformEffect({
     decode: urlString => Effect.try({
         try: () => {
             const url = new URL(urlString);
